iceoryx2-ffi/ffi/src/api/sample_mut_uninit.rs

```diff
@@ -27,21 +27,21 @@ use super::UninitPayloadFfi;
 
 // BEGIN types definition
 
-pub(super) union SampleMutUnion {
-    ipc: ManuallyDrop<SampleMut<ipc::Service, UninitPayloadFfi, UserHeaderFfi>>,
-    local: ManuallyDrop<SampleMut<local::Service, UninitPayloadFfi, UserHeaderFfi>>,
+pub(super) union SampleMutUninitUnion {
+    ipc: ManuallyDrop<SampleMutUninit<ipc::Service, UninitPayloadFfi, UserHeaderFfi>>,
+    local: ManuallyDrop<SampleMutUninit<local::Service, UninitPayloadFfi, UserHeaderFfi>>,
```

Comment:
If it's an either or on the C API, I think the going with just `SampleMut` makes more sense, else one has to call `iox2_sample_mut_uninit_send`, which is definitely the wrong message we want to spread.